orchestra/cmds/binary_archives.py

```diff
@@ -0,0 +1,89 @@
+import os
+
+from loguru import logger
+
+from . import CustomArgumentParser
+from .fix_binary_archives_symlinks import handle_fix_binary_archives_symlinks
+from .ls import handle_ls
+from ..actions.util import get_script_output, try_run_internal_script
+from ..gitutils import is_root_of_git_repo
+from ..model.configuration import Configuration
+
+_cmd_parser = None
+_subcmd_parser = None
+
+
+def install_subcommand(sub_argparser):
+    global _cmd_parser, _subcmd_parser
+    _cmd_parser = sub_argparser.add_parser("binary-archives",
+                                           handler=_handle_archives_top_level,
+                                           help="Manipulate binary archives")
+    _subcmd_parser = _cmd_parser.add_subparsers(
+        description="Available subcommands. Use <subcommand> --help",
+        dest="archives_command_name",
+        parser_class=CustomArgumentParser,
+        metavar="<subcommand>",
+    )
+    _subcmd_parser.add_parser(
+        "ls",
+        handler=handle_ls,
+        help="Print binary archives directories",
+    )
+    _subcmd_parser.add_parser(
+        "fix-symlinks",
+        handler=handle_fix_binary_archives_symlinks,
+        help="Fix symlinks in binary archives",
+    )
+    clean_subcmd = _subcmd_parser.add_parser(
+        "clean",
+        handler=handle_clean,
+        help="Delete stale binary archives"
+    )
+    clean_subcmd.add_argument(
+        "--pretend",
+        action="store_true",
+        help="Only print what would be done"
+    )
+
+
+def _handle_archives_top_level(args):
+    global _subcmd_parser
+    subcommand_parser = _subcmd_parser.choices.get(args.archives_command_name)
+    if subcommand_parser is None:
+        _cmd_parser.print_help()
+        exit(1)
+
+    subcommand_parser.handler(args)
+
+
+def handle_clean(args):
+    config = Configuration(use_config_cache=args.config_cache)
+    for name in config.binary_archives_remotes.keys():
+        path = os.path.join(config.binary_archives_dir, name)
+        if is_root_of_git_repo(path):
```

Comment:
Shouldn't we raise an error in case this is not the case?

Reply:
If the directory exists but is not the root of a git repo then maybe we should, if it does not exist at all then it's ok.
What do you think about this?
```python
if is_root_of_git_repo(path):
    ...
elif os.path.exists(path):
    logger.warning("...")
```

Reply:
Makes sense.